### PR TITLE
ignore wrong template on map

### LINKED ISSUE #10593: TD - GDI - Mission 3 crashes to desktop
**What you were doing when something broke?**
Loading Mission 3 of GDI on the Tiberium Dawn mod.

**What you expected to happen?**
The mission to load.

**What actually happened?**
Crash to desktop.

**If it's a crash, the contents of exception.log.**
```
Tiberian Dawn Mod at Version {DEV_VERSION}
on map c8adcbebb529bfbadfc2e96a634c532b9d1f9117 (Destroy The SAM Sites by Westwood Studios).
Operating System: Windows (Microsoft Windows NT 6.1.7601 Service Pack 1)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.Collections.Generic.KeyNotFoundException`: The given key was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at OpenRA.ReadOnlyDictionary`2.get_Item(TKey key) in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Primitives\ReadOnlyDictionary.cs:line 77
   at OpenRA.Graphics.TerrainRenderer.UpdateCell(CPos cell) in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Graphics\TerrainRenderer.cs:line 49
   at OpenRA.Graphics.TerrainRenderer..ctor(World world, WorldRenderer wr) in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Graphics\TerrainRenderer.cs:line 40
   at OpenRA.Graphics.WorldRenderer..ctor(World world) in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Graphics\WorldRenderer.cs:line 54
   at OpenRA.Game.StartGame(String mapUID, WorldType type) in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Game.cs:line 158
   at OpenRA.Network.UnitOrders.ProcessOrder(OrderManager orderManager, World world, Int32 clientId, Order order) in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Network\UnitOrders.cs:line 109
   at OpenRA.Network.OrderManager.TickImmediate() in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Network\OrderManager.cs:line 124
   at OpenRA.Sync.<>c__DisplayClass13_0.<CheckSyncUnchanged>b__0() in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Sync.cs:line 170
   at OpenRA.Sync.CheckSyncUnchanged[T](World world, Func`1 fn) in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Sync.cs:line 178
   at OpenRA.Sync.CheckSyncUnchanged(World world, Action fn) in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Sync.cs:line 170
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager) in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Game.cs:line 529
   at OpenRA.Game.LogicTick() in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Game.cs:line 582
   at OpenRA.Game.Loop() in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Game.cs:line 709
   at OpenRA.Game.Run() in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Game.cs:line 749
   at OpenRA.Program.Run(String[] args) in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Support\Program.cs:line 117
   at OpenRA.Program.Main(String[] args) in C:\cygwin64\home\Sam\OpenRA\OpenRA.Game\Support\Program.cs:line 39
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/30182667-td-gdi-mission-3-crashes-to-desktop?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by s7jones**
Fixes #10593.

I've implemented the changes suggested by @abcdefg30 to TerrainRenderer.cs.

This might be a bit of a brute force approach though?

### CODE REVIEW

**Comment by abcdefg30** [comment]
I'm not sure if just ignoring this is the right way, we should also remove the bogus template on the map.

Anyway, I can confirm that this fix works. :+1:

**Comment by pchote** [comment]
It would be better (i.e. retaining the behaviour from before the multi palette feature) to fall back to the default palette instead of returning.

**Comment by s7jones** [comment]
Sorry I thought this was no longer relevant (and closed it) due to another pull request but #10659 suggests this one is also needed (so reopened).

File: OpenRA.Game/Graphics/TerrainRenderer.cs [hunk 1]
```diff
@@ -46,6 +46,8 @@ public TerrainRenderer(World world, WorldRenderer wr)
 		public void UpdateCell(CPos cell)
 		{
 			var tile = mapTiles[cell];
+			if (!world.TileSet.Templates.ContainsKey(tile.Type))
+				return;
 			var palette = world.TileSet.Templates[tile.Type].Palette ?? TileSet.TerrainPaletteInternalName;
```
Comment (Mailaender) [inline OpenRA.Game/Graphics/TerrainRenderer.cs]: I guess this wants to be something like

```csharp
var palette = world.TileSet.Templates.ContainsKey(tile.Type) ?
	world.TileSet.Templates[tile.Type].Palette ?? TileSet.TerrainPaletteInternalName : 
	TileSet.TerrainPaletteInternalName;
```

**Comment by pchote** [comment]
This has been idle for a while, so I've incorporated the fixed check in #10897 (which already touches this same code).  Thanks!

**Pull request closed without merging**